components/trade/overviewModal/TradeOverviewModal.vue

```diff
@@ -195,33 +191,37 @@ const details = computed<Details & OverviewModeDetails>(() =>
         transactionSuccessTab: '',
       })
 
-const { data: nft, pending: nftLoading } = await useAsyncData<TradeNFTs | null>(`tarde-nft-id-${nftId.value}`, async () => {
-  if (!nftId.value) {
+const { data: nft, pending: nftLoading } = await useAsyncData<TradeNFTs | null>(`tarde-nft-id-${trade.value?.id}`, async () => {
+  if (!trade.value) {
     return null
   }
 
-  const [desired, offered] = await Promise.all([
-    useAsyncQuery<{ nftEntity: NFT }>({
-      query: nftById,
-      variables: {
-        id: nftId.value,
-      },
-      clientId: client.value,
-    }),
-    useAsyncQuery<{ nftEntity: NFT }>({
-      query: nftById,
-      variables: {
-        id: offeredItemId.value,
-      },
-      clientId: client.value,
-    }),
-  ])
+  const [offered, desired] = await Promise.all(
+    [
+      useAsyncQuery<{ nftEntity: NFT }>({
+        query: nftById,
+        variables: {
+          id: trade.value.offered.id,
+        },
+        clientId: client.value,
+      }),
+      trade.value.desired
+        ? useAsyncQuery<{ nftEntity: NFT }>({
+          query: nftById,
+          variables: {
+            id: trade.value.desired.id,
+          },
+          clientId: client.value,
+        })
+        : undefined,
+    ].filter(Boolean),
+  )
 
   return {
-    desired: desired.data.value.nftEntity,
     offered: offered.data.value.nftEntity,
```

Comment:
offered possibly undefined

Reply:
how is offered possibly undefined ?

Reply:
based on types above. for example if the query fails something happen on the query

```
const offered: _AsyncData<{
    nftEntity: NFT;
}, Error> | undefined
```

Reply:
unless, you remove the filter(Boolean) stuff

Reply:
changed 
99e90b6